src/kernel-install/kernel-install.c

```diff
@@ -427,6 +443,19 @@ static int context_set_initrds(Context *c, char* const* strv) {
         return context_set_path_strv(c, strv, "command line", "initrds", &c->initrds);
 }
 
+static int context_set_bls_type(Context *c, const char *s) {
+        assert(c);
+        BlsType b;
+        b = bls_type_from_string(s);
+        if (b >= 0)
+                c->arg_bls_type = b;
+        else {
+                log_error_errno(b, "Invalid bls-type: %s", s);
+                return _BLS_TYPE_INVALID;
```

Comment:
you can combine this into one line, i.e. `return log_error_errno(b, …` after all b is already _BLS_TYPE_INVALID (aka -EINVAL) here.